plugins/auth-backend/src/lib/resolvers/CatalogAuthResolverContext.ts

```diff
@@ -0,0 +1,176 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TokenManager } from '@backstage/backend-common';
+import { CatalogApi } from '@backstage/catalog-client';
+import {
+  DEFAULT_NAMESPACE,
+  Entity,
+  parseEntityRef,
+  RELATION_MEMBER_OF,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { ConflictError, InputError, NotFoundError } from '@backstage/errors';
+import { Logger } from 'winston';
+import { TokenIssuer } from '../..';
+import { TokenParams } from '../../identity';
+import { AuthResolverContext } from '../../providers';
+import { AuthResolverCatalogUserQuery } from '../../providers/types';
+import { CatalogIdentityClient } from '../catalog';
+
+/**
+ * Uses the default ownership resolution logic to return an array
+ * of entity refs that the provided entity claims ownership through.
+ *
+ * A reference to the entity itself will also be included in the returned array.
+ *
+ * @public
+ */
+export function getDefaultOwnershipEntityRefs(entity: Entity) {
+  const membershipRefs =
+    entity.relations
+      ?.filter(r => r.type === RELATION_MEMBER_OF)
+      .map(r => r.targetRef) ?? [];
+
+  return Array.from(new Set([stringifyEntityRef(entity), ...membershipRefs]));
+}
+
+/**
+ * @internal
+ */
+export class CatalogAuthResolverContext implements AuthResolverContext {
+  static create(options: {
+    logger: Logger;
+    catalogApi: CatalogApi;
+    tokenIssuer: TokenIssuer;
+    tokenManager: TokenManager;
+  }): CatalogAuthResolverContext {
+    const catalogIdentityClient = new CatalogIdentityClient({
+      catalogApi: options.catalogApi,
+      tokenManager: options.tokenManager,
+    });
+    return new CatalogAuthResolverContext(
+      options.logger,
+      options.tokenIssuer,
+      catalogIdentityClient,
+      options.catalogApi,
+      options.tokenManager,
+    );
+  }
+
+  private constructor(
+    public readonly logger: Logger,
+    public readonly tokenIssuer: TokenIssuer,
+    public readonly catalogIdentityClient: CatalogIdentityClient,
+    private readonly catalogApi: CatalogApi,
+    private readonly tokenManager: TokenManager,
+  ) {}
+
+  async issueToken(params: TokenParams) {
+    const token = await this.tokenIssuer.issueToken(params);
+    return { token };
+  }
+
+  async findCatalogUser(query: AuthResolverCatalogUserQuery) {
+    let result: Entity[] | Entity | undefined = undefined;
+    const { token } = await this.tokenManager.getToken();
+
+    if ('entityRef' in query) {
+      const entityRef = parseEntityRef(query.entityRef, {
+        defaultKind: 'user',
```

Comment:
no love for uppercase U kind? :) same below - i think we should always write them out as User, _especially_ if we then start shifting to retaining the case in the future...

Reply:
Yeah doesn't really matter atm does it 😅. Can switch it up if you prefer

Reply:
just how we generally do it i'd say and it matches the entity definitions so looks familiar to a reader, but no biggie